aiohttp_pydantic/injectors.py

```diff
@@ -69,6 +69,10 @@ async def inject(self, request: BaseRequest, args_view: list, kwargs_view: dict)
             raise HTTPBadRequest(
                 text='{"error": "Malformed JSON"}', content_type="application/json"
             ) from None
+        if not type(body) == dict:
+            raise HTTPBadRequest(
+                text='{"error": "Malformed JSON"}', content_type="application/json"
+            ) from None
 
         kwargs_view[self.arg_name] = self.model(**body)
```

Comment:
I think we can accept all pydantic model (list or dict) updating this line to

 kwargs_view[self.arg_name] = self.model.parse_obj(**body)

Reply:
Your suggestion works. I tried a slight variation on the demo, and also the OAS page looks good 🚀 
However, I've encountered what I _think_ is a bug in Pydantic's `parse_obj` method. See the failing test for explanation. Do you agree that this is a bug? If so, I'll open an issue.

Reply:
Yes for me is a bug, you can open an issue.
I tried with:

pydantic==1.7.3
pydantic==1.8
pydantic==1.8.1

The result is always the same :-/
```
>>> Pet.parse_obj([{"name": "cat", "age": 2}])
Pet(name='age', age=None)
```

Reply:
Ok I understood the issue, in fact `parse_obj` can accept a list of  2-tuple as a dict
```
>>> Pet.parse_obj([("name", "cat"), ("age", 2)])
Pet(name='cat', age=2)
```

Could you add this in the `__init__` of BodyGetter
``` python
# Because pydantic accepts a list of 2-tuples as dict or it raises a TypeError instead of a ValidationError
self._expect_object = self.model.schema()["type"] == "object"
```

and in the `BodyGetter.inject` raise an error HTTPBadRequest if `self._expect_object and not isinstance(body, dict)`

This bug/strange behaviour only exist when the pydantic.BaseModel is an object/dict. They are no issue with other type:

```
>>> class I(BaseModel):
...    __root__: int
... 

>>> I.parse_obj(3)
I(__root__=3)
>>> I.parse_obj([3])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pydantic/main.py", line 563, in pydantic.main.BaseModel.parse_obj
  File "pydantic/main.py", line 391, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for I
__root__
  value is not a valid integer (type=type_error.integer)
```

Reply:
Yes, better to be strict 👍🏼
There seem to be more issues around this, and they plan to make pydantic v2 more strict: https://github.com/samuelcolvin/pydantic/issues/1268